common/planners/src/plan_read_datasource.rs

```diff
@@ -27,16 +28,21 @@ use crate::Statistics;
 // TODO: Delete the scan plan field, but it depends on plan_parser:L394
 #[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq)]
 pub struct ReadDataSourcePlan {
+    // TODO encapsulate these 5 fields into TableInfo
     pub db: String,
     pub table: String,
     pub table_id: MetaId,
     pub table_version: Option<MetaVersion>,
     pub schema: DataSchemaRef,
+
     pub parts: Partitions,
     pub statistics: Statistics,
     pub description: String,
     pub scan_plan: Arc<ScanPlan>,
     pub remote: bool,
+
+    pub tbl_args: Option<Expression>,
```

Comment:
Maybe the type of `tbl_args` should be `Option<Vec<Expression>>`?

Reply:
I am not sure about this, just followed the original signature. 
If list of expressions should be represented as a vec of Expression instead of a variant of `Expression`, say sth like `Expression::List`, I'd like to change the signatures

Reply:
A subsequent "test drive" commit, which folds all the args into `TableArgs`

https://github.com/datafuselabs/databend/pull/1976/commits/4e9f8317a4f51918b0895dd0ed7654485a6f2dbc

If the last two commits are not ok, please just let me know, I 'd like to drop them from this PR, they are just experiments :D

Reply:
Seems we don't support table function with multiple arguments now.

Let's just keep it and fix it later.